api/server.spec.js

```diff
@@ -0,0 +1,50 @@
+const supertest = require('supertest')
+const server = require('./server');
+const db = require('../database/dbConfig');
+
+beforeEach(async () => {
+  // this function executes and clears out the table before each test
+  await db('users').truncate();
+});
+
+test('welcome route', async () => {
+  const res = await supertest(server).get('/')
+    expect(res.status).toBe(200)
+    expect(res.type).toBe('application/json')
+    expect(res.body.message).toMatch(/Sprint Challenge Authentication/i)
+});
+
+test('add a user to the database', async () => {
+  const res = await supertest(server)
+    .post('/api/auth/register')
+    .send({
+      "username": "test",
+      "password": "test"
+    })
+  expect(res.status).toBe(201)
+  expect(res.type).toBe('application/json')
+  expect(res.body.username).toBeString()
+});
+
+test('login as a test user and return a token', async () => {
+  const res = await supertest(server)
+    .post('/api/auth/login')
```

Comment:
you have a beforeEach which is truncating all the users if you are seeding.  Because of that, when you're logging in, this test shouldn't work because there's no user `test`.  You need to register `test` first and then you can login to make it work